store/sqlstore/integrity.go

```diff
@@ -0,0 +1,504 @@
+}
+
+func checkUsersOAuthAppsIntegrity(dbmap *gorp.DbMap) store.IntegrityCheckResult {
+	var config relationalCheckConfig
+	config.parentName = "Users"
+	config.parentIdAttr = "CreatorId"
+	config.childName = "OAuthApps"
+	config.childIdAttr = "Id"
+	return checkParentChildIntegrity(dbmap, config)
+}
+
+func checkUsersOAuthAuthDataIntegrity(dbmap *gorp.DbMap) store.IntegrityCheckResult {
+	var config relationalCheckConfig
+	config.parentName = "Users"
+	config.parentIdAttr = "UserId"
+	config.childName = "OAuthAuthData"
+	config.childIdAttr = "Code"
+	return checkParentChildIntegrity(dbmap, config)
+}
+
+func checkUsersOutgoingWebhooksIntegrity(dbmap *gorp.DbMap) store.IntegrityCheckResult {
+	var config relationalCheckConfig
+	config.parentName = "Users"
+	config.parentIdAttr = "CreatorId"
+	config.childName = "OutgoingWebhooks"
+	config.childIdAttr = "Id"
+	return checkParentChildIntegrity(dbmap, config)
+}
+
+func checkUsersPostsIntegrity(dbmap *gorp.DbMap) store.IntegrityCheckResult {
+	var config relationalCheckConfig
+	config.parentName = "Users"
+	config.parentIdAttr = "UserId"
+	config.childName = "Posts"
+	config.childIdAttr = "Id"
+	return checkParentChildIntegrity(dbmap, config)
+}
+
+func checkUsersPreferencesIntegrity(dbmap *gorp.DbMap) store.IntegrityCheckResult {
+	var config relationalCheckConfig
+	config.parentName = "Users"
+	config.parentIdAttr = "UserId"
+	config.childName = "Preferences"
+	return checkParentChildIntegrity(dbmap, config)
+}
+
+func checkUsersReactionsIntegrity(dbmap *gorp.DbMap) store.IntegrityCheckResult {
+	var config relationalCheckConfig
+	config.parentName = "Users"
+	config.parentIdAttr = "UserId"
+	config.childName = "Reactions"
+	return checkParentChildIntegrity(dbmap, config)
+}
+
+func checkUsersSessionsIntegrity(dbmap *gorp.DbMap) store.IntegrityCheckResult {
+	var config relationalCheckConfig
+	config.parentName = "Users"
+	config.parentIdAttr = "UserId"
+	config.childName = "Sessions"
+	config.childIdAttr = "Id"
+	return checkParentChildIntegrity(dbmap, config)
+}
+
+func checkUsersStatusIntegrity(dbmap *gorp.DbMap) store.IntegrityCheckResult {
+	var config relationalCheckConfig
+	config.parentName = "Users"
+	config.parentIdAttr = "UserId"
+	config.childName = "Status"
+	return checkParentChildIntegrity(dbmap, config)
+}
+
+func checkUsersTeamMembersIntegrity(dbmap *gorp.DbMap) store.IntegrityCheckResult {
+	var config relationalCheckConfig
+	config.parentName = "Users"
+	config.parentIdAttr = "UserId"
+	config.childName = "TeamMembers"
+	return checkParentChildIntegrity(dbmap, config)
+}
+
+func checkUsersUserAccessTokensIntegrity(dbmap *gorp.DbMap) store.IntegrityCheckResult {
+	var config relationalCheckConfig
+	config.parentName = "Users"
+	config.parentIdAttr = "UserId"
+	config.childName = "UserAccessTokens"
+	config.childIdAttr = "Id"
+	return checkParentChildIntegrity(dbmap, config)
+}
+
+func checkChannelsIntegrity(dbmap *gorp.DbMap, results chan<- store.IntegrityCheckResult) {
```

Comment:
It appears that we run the integrity checks in parallel -- is there any reason to accept a channel instead of just returning the results directly?

Reply:
I believe all those checks are running sequentially since they are part of the same group of checks (relational integrity), `store/sqlstore/supplier.go:1069` for reference. 

The choice of using a channel instead of returning results directly was dictated by the fact that I wanted the tool to be able to consume the reports as they were produced without waiting for everything to be completed. The other option was to pass a `io.Writter` but decided that using a channel was the more elegant approach.

Reply:
Oops, I indeed meant to write `in sequence` not `in parallel` -- I agree with you on that front.

I'd love to see some parallelization within these checks, especially if in a high-availability server with multiple read replicas to improve the performance of the integrity check. But that can be a future improvement.

Reply:
Yes, in the future it makes sense to add some parallelization to execute the checks more efficiently on multi-core systems.  
I made it granular enough by design so that it shouldn't be hard to implement alongside the ability to run a user defined subset of all the available checks.